### PR TITLE
Ctctraders 2195 - T2F previous administrative references logic

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

File: app/models/journeyDomain/PreviousReferences.scala [hunk 1]
```diff
@@ -50,9 +50,11 @@ object PreviousReferences {
...
     (
       DeclarationTypePage.reader,
-      CountryOfDispatchPage.reader
+      IsNonEuOfficePage.reader
     ).tupled.flatMap {
-      case (Option2, CountryOfDispatch(_, true)) =>
+      case (Option2, true) =>
+        allPreviousReferencesReader(itemIndex)
+      case (Option3, true) =>
         allPreviousReferencesReader(itemIndex)
```
Comment (ericohwotu) [inline app/models/journeyDomain/PreviousReferences.scala:58]: are Option 2 and 3 the same thing?

Comment (kevincoatsworth) [reply app/models/journeyDomain/PreviousReferences.scala:58]: They are separate things but have the same outcome, should we double them up on one case?

File: app/models/journeyDomain/PreviousReferences.scala [hunk 2]
```diff
@@ -50,9 +50,11 @@ object PreviousReferences {
   def derivePreviousReferences(itemIndex: Index): UserAnswersReader[Option[NonEmptyList[PreviousReferences]]] =
     (
       DeclarationTypePage.reader,
-      CountryOfDispatchPage.reader
+      IsNonEuOfficePage.reader
```
Comment (ericohwotu) [inline app/models/journeyDomain/PreviousReferences.scala:53]: Where is this set?

File: app/models/journeyDomain/PreviousReferences.scala (hunk 1 above)
Comment (ericohwotu) [reply app/models/journeyDomain/PreviousReferences.scala:58]: yees plese

File: app/models/journeyDomain/PreviousReferences.scala (hunk 2 above)
Comment (ericohwotu) [reply app/models/journeyDomain/PreviousReferences.scala:53]: Nevermind, ignore this

Comment (kevincoatsworth) [reply app/models/journeyDomain/PreviousReferences.scala:53]: It's set in the office of departure controller in the pretasklist section

**Review by ericohwotu — APPROVED** [review]

**Pull request merged**